sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/options/ProxyInvocationHandler.java

```diff
@@ -209,6 +261,69 @@ public int hashCode() {
   }
 
   /**
+   * Populate display data. See {@link HasDisplayData#populateDisplayData}.
+   */
+  private void populateDisplayData(DisplayData.Builder builder) {
+    Preconditions.checkState(canPopulateDisplayData(),
+        "Populating display data from deserialized PipelineOptions is not supported.");
+    Set<PipelineOptionsReflector.Property> props =
+        PipelineOptionsReflector.collectVisibleProperties(knownInterfaces);
+
+    Multimap<String, Class<?>> keyToInterfaceMap = buildOptionToInterfaceMap(props);
+
+    for (Map.Entry<String, BoundValue> option : options.entrySet()) {
+      BoundValue boundValue = option.getValue();
+      if (boundValue.isDefault()) {
+        continue;
+      }
+
+      for (Class<?> definingClass : keyToInterfaceMap.get(option.getKey())) {
+        Object value = boundValue.getValue() == null ? "" : boundValue.getValue();
+        builder.add(option.getKey(), value)
+           .withNamespace(definingClass);
+      }
+    }
+  }
+
+  private boolean canPopulateDisplayData() {
```

Comment:
I'd suggest inlining this. Otherwise, it's not completely clear the relation between `canPopulateDisplayData` and the error message which talks about deserialized `PipelineOptions`.